### ISSUE TITLE
a specific case 

### ISSUE THREAD

**Issue by no-on3** [machocam/rummikub#3]
Hello, 
I worked on this algo recently, my solution was initially based on your solution, with a bit of recursivity   
I added it because  some cases are not handled correctly 
for example : 

` [3, "A"],    [4, "A"],    [5, "A"],    [6, "A"],    [7, "A"], [6, "B"],    [6, "C"]` 

will return `[6, "B"],    [6, "C"]` as rest while it should be` [7, "A"] `only 

this is my solution, what do you think? 
https://github.com/no-on3/RummiKubPython/blob/main/script.py

**Comment by machocam** [comment]
Hey there,

Thanks a lot for your interest on this topic! It's a cool game, it's not
easy to write something which can play it well though.

I haven't really gone into the specifics of this in a few years - if I have
some time over the coming weeks I might sit down to see how I could improve
it. Overall though - as you can see in the style used in writing it ; I'm
no pro!

**Comment by no-on3** [comment]
yes, the algo is quite simple actually but the implementation with optimization is a bit hard!
take your time, thank you very much!
